internal/provider/resource_nd_site.go

```diff
@@ -0,0 +1,491 @@
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Read of resource nd_site with id '%s'", data.Id.ValueString()))
+
+	getAndSetSiteAttributes(ctx, &resp.Diagnostics, r.client, data)
+
+	// Save updated data into Terraform state
+	if data.Id.IsNull() {
+		var emptyData *SiteResourceModel
+		resp.Diagnostics.Append(resp.State.Set(ctx, &emptyData)...)
+	} else {
+		resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("End read of resource nd_site with id '%s'", data.Id.ValueString()))
+}
+
+func (r *SiteResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	tflog.Debug(ctx, "Start update of resource: nd_site")
+
+	var stateData *SiteResourceModel
+	var data *SiteResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateData)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Update of resource nd_site with id '%s'", data.Id.ValueString()))
+
+	jsonPayload := getSiteCreateJsonPayload(ctx, &resp.Diagnostics, data)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	DoRestRequest(ctx, &resp.Diagnostics, r.client, fmt.Sprintf("%s/%s", sitePath, data.Id.ValueString()), "PUT", jsonPayload)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	setSiteId(ctx, data)
+
+	getAndSetSiteAttributes(ctx, &resp.Diagnostics, r.client, data)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	tflog.Debug(ctx, "End update of resource nd_site")
+}
+
+func (r *SiteResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	tflog.Debug(ctx, "Start delete of resource: nd_site")
+	var data *SiteResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Delete of resource nd_site with id '%s'", data.Id.ValueString()))
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	DoRestRequest(ctx, &resp.Diagnostics, r.client, fmt.Sprintf("%s/%s", sitePath, data.Id.ValueString()), "DELETE", nil)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	tflog.Debug(ctx, fmt.Sprintf("End delete of resource nd_site with id '%s'", data.Id.ValueString()))
+}
+
+func (r *SiteResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	tflog.Debug(ctx, "Start import state of resource: nd_site")
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
+	var stateData *SiteResourceModel
+	resp.Diagnostics.Append(resp.State.Get(ctx, &stateData)...)
+
+	username := os.Getenv("ND_SITE_USERNAME")
+	if username == "" {
+		resp.Diagnostics.AddError("Missing input", "A username must be provided during import, please set the ND_SITE_USERNAME environment variable")
```

Comment:
->
"The username of the ND site must be provided during import, please set the ND_SITE_USERNAME environment variable"